homeassistant/components/goalzero/switch.py

```diff
@@ -0,0 +1,66 @@
+"""Support for Goal Zero Yeti Switches."""
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.const import CONF_NAME
+
+from . import YetiEntity
+from .const import DATA_KEY_API, DATA_KEY_COORDINATOR, DOMAIN, SWITCH_DICT
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Goal Zero Yeti switch."""
+    name = entry.data[CONF_NAME]
+    goalzero_data = hass.data[DOMAIN][entry.entry_id]
+    switches = [
+        YetiSwitch(
+            goalzero_data[DATA_KEY_API],
+            goalzero_data[DATA_KEY_COORDINATOR],
+            name,
+            switch_name,
+            entry.entry_id,
+        )
+        for switch_name in SWITCH_DICT
+    ]
+    async_add_entities(switches, True)
+
+
+class YetiSwitch(YetiEntity, SwitchEntity):
+    """Representation of a Goal Zero Yeti switch."""
+
+    def __init__(self, api, coordinator, name, switch_name, server_unique_id):
+        """Initialize a Goal Zero Yeti switch."""
+        super().__init__(api, coordinator, name, server_unique_id)
```

Comment:
Pass the device class to the parent since it's setting that property.

Reply:
Do you mean include it in the super()?

Reply:
Yes.